Tests failing on GPU

# Issue
## Problem Description

Related to Release 0.4.0, can be fixed directly on `release` branch.

Currently the tests fail on GPUs, for torch because of a missing transfer from GPU memory to host. For TF there seems to be a breaking API change ( ` ImportError: cannot import name 'np_config' from 'tensorflow.python.ops.numpy_ops'` , see https://stackoverflow.com/questions/75727569/cannot-import-name-np-config-from-tensorflow-python-ops-numpy-ops ) 

Logs:

pytorch_gpu: [pytest_gpu0.log](https://github.com/esa/torchquad/files/11500705/pytest_gpu0.log)


TF_gpu, torch_cpu (failed to get JAX working since this was on a win machine): [pytest_all_gpu0.log](https://github.com/esa/torchquad/files/11500709/pytest_all_gpu0.log)


Setting up an env to check with all frameworks on GPU also proved time-consuming. (and failed for jax)

## Expected Behavior

- Tests passing.
- (optional) Easier way to check in the future

## What Needs to be Done
- [x] Add calls to transfer to host memory
- [x] Fix / update TF API call
- [x] (optional) add another .yml for GPU version of all libs.

## How Can It Be Tested or Reproduced

Run pytest , used following env

```
name: torchquad_all
channels:
  - anaconda
  - conda-forge
  - pytorch
  - nvidia
dependencies:
  - autoray>=0.2.5
  - loguru>=0.5.3
  - matplotlib>=3.3.3
  - pytest>=6.2.1
  - python>=3.8
  - scipy>=1.6.0
  - sphinx>=3.4.3
  - tqdm>=4.56.0
  # Numerical backend installations with CUDA support where possible:
  - numpy>=1.19.5
  - pytorch>=1.9 # CPU version
  - tensorflow-gpu
  # jaxlib with CUDA support is not available for conda
  - pip:
      - --find-links https://storage.googleapis.com/jax-releases/jax_releases.html
      - jax[cpu]>=0.2.22 # this will only work on linux. for win see e.g. https://github.com/cloudhan/jax-windows-builder
        # CPU version
```

@ilan-gold running your notebook in colab, I get

```
FAILED gauss_test.py::test_integrate_torch - assert (3 > 3 or 7.105427357601002e-15 < 2e-16)
FAILED gauss_test.py::test_integrate_tensorflow - assert (3 > 3 or 7.105427357601002e-15 < 2e-16)
```
Was this what you got?

On one of our GPU servers I get

```
FAILED gauss_test.py::test_integrate_torch - assert (3 > 3 or 7.105427357601002e-15 < 2e-16)
FAILED gauss_test.py::test_integrate_tensorflow - assert (3 > 3 or 7.105427357601002e-15 < 2e-16)
FAILED gradient_test.py::test_gradients_torch - AssertionError: assert 0.11964358930767993 < 0.1
```
so seems the test bounds are bit too harsh for gauss and the gradient test?

I also get 115 warnings, oof. We might wanna look at those at some point :D :see_no_evil:

--- a/torchquad/tests/gauss_test.py
+++ b/torchquad/tests/gauss_test.py
@@ -44,7 +44,7 @@ def _run_gaussian_tests(backend, _precision):
     # All polynomials up to degree 3 = 2N-1 should be 0, others should be good as well.
     # If this breaks check if test functions in helper_functions changed.
     for err, test_function in zip(errors, funcs):
-        assert test_function.get_order() > 3 or err < 2e-16
+        assert test_function.get_order() > 3 or err < 8e-15
     for error in errors[:2]:
         assert error < 1e-15
 

--- a/torchquad/tests/gradient_test.py
+++ b/torchquad/tests/gradient_test.py
@@ -264,7 +264,7 @@ def _run_gradient_tests(backend, dtype_name):
         )
         # Check if the integral and gradient are accurate enough
         assert np.abs(integral - 34.0) < 0.2
-        assert np.abs(gradient - 4.0) < 0.1
+        assert np.abs(gradient - 4.0) < 0.15
 
 
 test_gradients_torch = setup_test_for_backend(_run_gradient_tests, "torch", "float64")

